### PR TITLE
gateway-api: Add supported features in GatewayClass status

### PR DESCRIPTION
**Pull request by sayboras**
There is a new status.supportedFeature introduced in Gateway resource, the goal is to make things visible to users about capabilities of implementation.  This commit is to add all the supported features into cilium-managed Gateway. 

Kindly note that the list is hard coded right now, as there is no enum/const defined in upstream main codebase (only available in conformance tests).

### CODE REVIEW

File: operator/pkg/gateway-api/gatewayclass_reconcile.go [hunk 1]
```diff
@@ -46,6 +46,29 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// Hence, just set gateway class Accepted condition to true blindly.
 	setGatewayClassAccepted(gwc, true)
 
+	// List of features supported by Cilium.
+	// The same is used in GHA CI .github/workflows/conformance-gateway-api.yaml
```
Comment (meyskens) [inline operator/pkg/gateway-api/gatewayclass_reconcile.go:50]: the .github seems weird here?

**Comment by meyskens** [comment]
Can you add a PR description :) changes look good but probably needs some context to merge it

File: operator/pkg/gateway-api/gatewayclass_reconcile.go (hunk 1 above)
Comment (sayboras) [reply operator/pkg/gateway-api/gatewayclass_reconcile.go:50]: ah it's the full path from root directory to gha workflow file

Comment (meyskens) [reply operator/pkg/gateway-api/gatewayclass_reconcile.go:50]: :facepalm:

**Review by meyskens — APPROVED** [review]
LGTM ship it!

**Comment by levikobi** [comment]
Just for reference in the GEP, this relates to https://github.com/kubernetes-sigs/gateway-api/issues/2162

**Comment by LiorLieberman** [comment]
@meyskens @sayboras I think this should be identical to the features set used in the conformance tests. 

Also, looks like it is not covered by any tests.

**Comment by sayboras** [comment]
> I think this should be identical to the features set used in the conformance tests.

You are correct. I was waiting for the upstream conformance test to run tests depending on the feature sets in the gateway status. However, there are still some gaps upstream, so this is manually updated right now.

**Pull request merged**